packages/terra-consumer-nav/src/components/modal/Modal.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import IconClose from 'terra-icon/lib/icon/IconClose';
+import Overlay from 'terra-overlay';
+import styles from './Modal.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Title to be rendered on top of the modal content
+   */
+  title: PropTypes.string,
+  /**
+   * Content to be rendered in the body of the dialog box.
+   */
+  content: PropTypes.element,
+  /**
+   * Boolean value to render modal.
+   */
+  isModalOpen: PropTypes.bool.isRequired,
+  /**
+   * Required callback function for use by parent component to update state.
+   */
+
+  closeModal: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  title: null,
+  content: null,
```

Comment:
@Mahmud-Khan can you address this issue across other components, not just this one (like nav.jsx)